activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -432,6 +432,7 @@ def in_order_of(column, values)
       references = column_references([column])
       self.references_values |= references unless references.empty?
 
+      values = values.map { |value| type_caster.type_cast_for_database(column, value) }
```

Comment:
Minor: what about using `map!` instead? Should be a minor performance improvement

Reply:
@intrip thank you very much for the review! 
I updated the PR to use `map!` instead of `map`.